crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -1824,26 +1834,28 @@ impl StateStore for SqliteStateStore {
         // Note: ROWID is always present and is an auto-incremented integer counter. We
         // want to maintain the insertion order, so we can sort using it.
         // Note 2: transaction_id is not encoded, see why in `save_send_queue_event`.
-        let res: Vec<(String, Vec<u8>, Option<Vec<u8>>, usize)> = self
+        let res: Vec<(String, Vec<u8>, Option<Vec<u8>>, usize, u64)> = self
             .acquire()
             .await?
             .prepare(
-                "SELECT transaction_id, content, wedge_reason, priority FROM send_queue_events WHERE room_id = ? ORDER BY priority DESC, ROWID",
+                "SELECT transaction_id, content, wedge_reason, priority, enqueue_time FROM send_queue_events WHERE room_id = ? ORDER BY priority DESC, ROWID",
                 |mut stmt| {
                     stmt.query((room_id,))?
-                        .mapped(|row| Ok((row.get(0)?, row.get(1)?, row.get(2)?, row.get(3)?)))
+                        .mapped(|row| Ok((row.get(0)?, row.get(1)?, row.get(2)?, row.get(3)?, row.get(4)?)))
                         .collect()
                 },
             )
             .await?;
 
         let mut requests = Vec::with_capacity(res.len());
         for entry in res {
+            let enqueue_time = MilliSecondsSinceUnixEpoch(UInt::new(entry.4).unwrap());
```

Comment:
We can't use `unwrap()` here, we'd need to return an error if the `UInt` call failed.

Reply:
Do you think dropping the malformed data is reasonable fallback if it fails to unpack the data?
```
  let enqueue_time = UInt::new(entry.4).map(|e| MilliSecondsSinceUnixEpoch(e));
```

Reply:
Yeah, it's a good idea to keep it `Option`al :+1: